### PR TITLE
Adds most ios snippets

### PR DESCRIPTION
**Pull request by andrewxhill**
Signed-off-by: andrew <andrewxhill@gmail.com>

Probably a bit more iOS folks with AltConf tweets. Also need this stuff to prepare my talk anyway.

This is all just ObjC obv, but still need Swift examples for #34

### CODE REVIEW

**Comment by andrewxhill** [comment]
bump @asutula

File: snippets/common.yml [hunk 1]
```diff
@@ -230,7 +275,14 @@ contacts:
     react_native:
       code: '@todo'
     ios:
-      code: '@todo'
+      code: |
+            NSError *error;
+            QueryOptions *options;
```
Comment (asutula) [inline snippets/common.yml]: This needs to be initialized, it's just a null pointer like this, will lead to runtime crash

File: snippets/common.yml [hunk 2]
```diff
@@ -244,7 +296,14 @@ contacts:
...
```
Comment (asutula) [inline snippets/common.yml]: Same, initialize.

File: snippets/common.yml [hunk 3]
```diff
@@ -244,7 +296,14 @@ contacts:
...
+            options.wait = 10;
+            options.limit = 1;
+            ContactQuery *query;
```
Comment (asutula) [inline snippets/common.yml]: Init.

File: snippets/common.yml [hunk 4]
```diff
@@ -230,7 +275,14 @@ contacts:
...
```
Comment (asutula) [inline snippets/common.yml]: Init.

File: snippets/common.yml [hunk 5]
```diff
@@ -258,7 +317,9 @@ contacts:
     react_native:
       code: '@todo'
     ios:
-      code: '@todo'
+      code: |
+            Contact searchResult = // Should be a Contact object as returned by contact search
```
Comment (asutula) [inline snippets/common.yml]: This should be a pointer with `*`

File: snippets/common.yml [hunk 6]
```diff
@@ -454,7 +567,12 @@ files:
             const result = await textile.files.prepare(input, "12D3KooWSYT6SUL9fx15pwjHSVUsuymnbixmRtPGySmFYtWE51Sc")
             const block = await textile.files.add(result.dir, "12D3KooWSYT6SUL9fx15pwjHSVUsuymnbixmRtPGySmFYtWE51Sc")
     ios:
-      code: '@todo'
+      code: |
+            NSError *error;
+            NSString *threadId = @"12D3KooWSYT6SUL9fx15pwjHSVUsuymnbixmRtPGySmFYtWE51Sc";
+            NSString *filePath = @"/path/to/mmm-bytes....";
+            MobilePreparedFiles *files = [Textile.instance.files prepareByPathSync:filePath threadId:threadId error:&error];
```
Comment (asutula) [inline snippets/common.yml]: Again, maybe the async version is better here.

File: snippets/common.yml [hunk 7]
```diff
@@ -471,7 +589,12 @@ files:
             const result = await textile.files.prepare(base64ImageString, "12D3KooWNihfHDLsiJ36qQRQQ2vMcBZRHd2VBs3wTFtyGEk7zGeq")
             const block = await textile.files.add(result.dir, "12D3KooWNihfHDLsiJ36qQRQQ2vMcBZRHd2VBs3wTFtyGEk7zGeq")
     ios:
-      code: '@todo'
+      code: |
+            NSError *error;
+            NSString *threadId = @"12D3KooWSYT6SUL9fx15pwjHSVUsuymnbixmRtPGySmFYtWE51Sc";
+            NSString *filePath = @"/path/to/image";
+            MobilePreparedFiles *files = [Textile.instance.files prepareByPathSync:filePath threadId:threadId error:&error];
```
Comment (asutula) [inline snippets/common.yml]: And here.

File: docs/a-tour-of-textile.md [hunk 8]
```diff
@@ -2941,6 +2954,10 @@ Downloading data behind a path is one of the most useful IPFS APIs. For example,
 {{ipfs.cat.js_http_client.code}}
 ```
 
+```Swift tab="iOS"
```
[the remark this answers is not in the document]
Comment (andrewxhill) [reply docs/a-tour-of-textile.md]: trying to figure that out in the other branch now

**Pull request merged**